frontend/pages/hosts/HostDetailsPage/SoftwareTable/SoftwareTableConfig.tsx

```diff
@@ -0,0 +1,137 @@
+import React from "react";
+
+import HeaderCell from "components/TableContainer/DataTable/HeaderCell/HeaderCell";
+import TextCell from "components/TableContainer/DataTable/TextCell";
+import IconTooltipCell from "components/TableContainer/DataTable/IconTooltipCell";
+import { ISoftware } from "interfaces/software";
+
+interface IHeaderProps {
+  column: {
+    title: string;
+    isSortedDesc: boolean;
+  };
+}
+
+interface ICellProps {
+  cell: {
+    value: any;
+  };
+  row: {
+    original: ISoftware;
+  };
+}
+
+interface IDataColumn {
+  title: string;
+  Header: ((props: IHeaderProps) => JSX.Element) | string;
+  accessor: string;
+  Cell: (props: ICellProps) => JSX.Element;
+  disableHidden?: boolean;
+  disableSortBy?: boolean;
+}
+
+interface ISoftwareTableData extends ISoftware {
+  type: string;
+}
+
+// NOTE: cellProps come from react-table
+// more info here https://react-table.tanstack.com/docs/api/useTable#cell-properties
+const generateTableHeaders = (): IDataColumn[] => {
+  return [
+    {
+      title: "Vulnerabilities",
+      Header: "",
+      disableSortBy: true,
+      accessor: "vulnerabilitiesTooltip",
+      Cell: (cellProps) => <IconTooltipCell value={cellProps.cell.value} />,
+    },
+    {
+      title: "Name",
+      Header: (cellProps) => (
+        <HeaderCell
+          value={cellProps.column.title}
+          isSortedDesc={cellProps.column.isSortedDesc}
+        />
+      ),
+      accessor: "name",
+      Cell: (cellProps) => <TextCell value={cellProps.cell.value} />,
+    },
+    {
+      title: "Type",
+      Header: (cellProps) => (
+        <HeaderCell
+          value={cellProps.column.title}
+          isSortedDesc={cellProps.column.isSortedDesc}
+        />
+      ),
+      disableSortBy: false,
+      accessor: "type",
+      Cell: (cellProps) => <TextCell value={cellProps.cell.value} />,
+    },
+    {
+      title: "Installed Version",
+      Header: "Installed Version",
+      disableSortBy: true,
+      accessor: "version",
+      Cell: (cellProps) => <TextCell value={cellProps.cell.value} />,
+    },
+  ];
+};
+
+const TYPE_CONVERSION: any = {
```

Comment:
Let's try to use more specific types:
```suggestion
const TYPE_CONVERSION: Record<string, string> = {
```

Reply:
Thank you! I couldn't quite figure this one out. If you have any clear-cut TypeScript type resource I could bookmark, that would be amazing.